src/model/mod.rs

```diff
@@ -10,6 +10,54 @@
 //! The raw version can be converted into the owned version by cloning all the fields. The owned
 //! version can produce a raw version by borrowing all fields (roughly speaking).
 
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
 pub mod creator;
 pub mod level;
 pub mod song;
+
+/// Enum modelling the version of a Geometry Dash client
+#[derive(Debug, Clone, Copy, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum GameVersion {
+    /// Variant representing an unknown version. This variant is only used for
+    /// levels that were uploaded before the game started tracking the
+    /// version. This variant's string
+    /// representation is `"10"`
+    Unknown,
+
+    /// Variant representing a the version represented by the given minor/major
+    /// values in the form `major.minor`
+    Version { minor: u8, major: u8 },
+}
+
+impl From<u8> for GameVersion {
+    fn from(version: u8) -> Self {
+        if version == 10 {
+            GameVersion::Unknown
+        } else {
+            GameVersion::Version {
+                major: (version / 10) as u8,
+                minor: (version % 10) as u8,
+            }
+        }
+    }
+}
+
+impl Into<u8> for GameVersion {
```

Comment:
We shouldn't implement Into directly (as Rust docs suggest), we can implement From<GameVersion> for u8

Reply:
Oh, I did not know that!